tests/service.rs

```diff
@@ -24,8 +24,38 @@ fn service_should_be_started_successfully() {
                 - key: kubernetes.io/arch
                   operator: Equal
                   value: stackable-linux
-        "},
+        "}),
     );
 
     client.verify_pod_condition(&pod, "Ready");
 }
+
+#[test]
+fn restart_after_ungraceful_shutdown_should_succeed() {
```

Comment:
super nit: does not necessarily have to be an "ungraceful" shutdown, a normal shutdown taking longer than terminationGracePeriod during a restart would be sufficient to trigger the bug.

Reply:
The termination grace period actually has nothing to do with this bug. The bug occurs when the shutdown is not finished until the agent tries to start the new systemd service. So it can occur if the shutdown takes longer than one or two seconds. To reproduce the bug it is necessary to set a duration for the shutdown which is achieved with an ungraceful shutdown after the termination grace period. So the function name describes the test case but not the bug.